pkg/controller/common.go

```diff
@@ -1744,47 +1742,29 @@ func objectDeletionRequested(obj runtime.Object) bool {
 	return !deletionTime.IsZero() && deletionTime.Time.Before(time.Now())
 }
 
-func isCleanupRequested(obj runtime.Object) bool {
-	meta, _ := meta.Accessor(obj)
-	if meta == nil {
-		return false
-	}
-	annotations := meta.GetAnnotations()
-	if annotations == nil {
-		return false
-	}
-	_, requested := annotations[consts.CleanUpAnnotation]
-	return requested
+func isCleanupRequested(attachment *diskv1beta1.AzVolumeAttachment) bool {
+	return attachment != nil && attachment.Status.Annotations != nil && contains(attachment.Status.Annotations, consts.CleanUpAnnotation)
 }
 
 func volumeDetachRequested(attachment *diskv1beta1.AzVolumeAttachment) bool {
-	return attachment != nil && attachment.Annotations != nil && metav1.HasAnnotation(attachment.ObjectMeta, consts.VolumeDetachRequestAnnotation)
+	return attachment != nil && attachment.Status.Annotations != nil && contains(attachment.Status.Annotations, consts.VolumeDetachRequestAnnotation)
 }
 
 func volumeDeleteRequested(volume *diskv1beta1.AzVolume) bool {
-	return volume != nil && volume.Annotations != nil && metav1.HasAnnotation(volume.ObjectMeta, consts.VolumeDeleteRequestAnnotation)
+	return volume != nil && volume.Status.Annotations != nil && contains(volume.Status.Annotations, consts.VolumeDeleteRequestAnnotation)
 }
 
 func isDemotionRequested(attachment *diskv1beta1.AzVolumeAttachment) bool {
 	return attachment != nil && attachment.Status.Detail != nil && attachment.Status.Detail.Role == diskv1beta1.PrimaryRole && attachment.Spec.RequestedRole == diskv1beta1.ReplicaRole
 }
 
 func isPreProvisionCleanupRequested(volume *diskv1beta1.AzVolume) bool {
-	return volume != nil && volume.Annotations != nil && metav1.HasAnnotation(volume.ObjectMeta, consts.PreProvisionedVolumeCleanupAnnotation)
-}
-
-func finalizerExists(finalizers []string, finalizerName string) bool {
-	for _, finalizer := range finalizers {
-		if finalizer == finalizerName {
-			return true
-		}
-	}
-	return false
+	return volume != nil && volume.Status.Annotations != nil && contains(volume.Status.Annotations, consts.PreProvisionedVolumeCleanupAnnotation)
 }
 
-func labelExists(labels map[string]string, label string) bool {
-	if labels != nil {
-		_, ok := labels[label]
+func contains(mmap map[string]string, key string) bool {
```

Comment:
Consider `statusAnnotationsContains`

Reply:
Turned it into a generic function called `mapContains` for `map[string]string`